### DISCUSSION TITLE
daemon API: provide command to inform on trading and mining fees when taking an offer

### DISCUSSION THREAD

**Discussion by pmartincalvo** [bisq-network/bisq#6332] [Ideas]
In bisq GUI, when you select an offer from the book to take it, you land on a confirmation screen. This screen, among other things, informs you about the trading and mining fees you will incur if you finally confirm and take the offer.

In the bisq daemon API there is no way to obtain this trading and mining fee before taking an offer. The response `OfferInfo` returned by methods `GetOffers` and `GetOffer` does not provide any information on the trading fees, mining fees and deposit requirements.

I believe that a command that provides this info is key for the bisq daemon. Otherwise, users blindly jump into trades unaware of these important details. This could be done by either creating a new command entirely or expanding the `GetOffer` method to include this information.

**Comment by ripcurlx** [comment]
@ghubstan ☝️

**Comment by ghubstan** [comment]
@pmartincalvo This does need to be fixed.  What would be your preference?

(a) Adding a second table of `CLI` output to the `getoffer` command, showing the trading and mining fee info?   Adding this type of info to the `getoffer` method results would make it inconsistent with the details you see when viewing offer details in the GUI -- where no fee details are displayed. 
	
(b) Adding a `--dryrun=true` param to the existing `takeoffer` command?  This would return the fee details instead of actually taking the offer.

(c) Adding a new method/command to the API.  `takeoffer-dryrun`?  `takeoffer-fees`?  I'm sure there's a better method name for this.

There may be other ways to implement this.

I prefer option (b), then option (c).

Option (b) would be analogous to clicking the `TAKE OFFER` button in the GUI and then looking at the `Fund your trade` pop-up.  This pop-up shows security deposit amt., trade amt., trading fee, total mining fees.  You do not see these details when you click an offer's details link.

**Reply by pmartincalvo** [reply]
I also don't like (a). It makes `GetOffer` stop being idempotent since the mining fees can change, which doesn't feel right.

I think both (b) or (c) could work. To be honest, I think (b) might be a bit obscure for freshers who are starting into bisq.

I guess if we find the right naming, (c) should be fine. `GetOfferTemplate`? `GetOfferFineprint`? `GetOfferConditions`?

Maybe the actual bisq GUI code that prepares the confirmation screen is a good source of inspiration. Sorry for not looking myself, but i don't have the knowledge to find that in the repo.

**Reply by ghubstan** [reply]
>> I guess if we find the right naming, (c) should be fine. 

I think we can come up with a good (new) method name.  Think of the CLI commands instead of the source code... there is `takeoffer` which we will leave as is (no new `--dryrun` param).  And I think we can eliminate `getoffer*` because this problem is about how much it costs to take an offer, not look at it.

How about a new `takeofferfees` API CLI command?   `seetakeofferfees` (too verbose).  
You have any suggestions?

**Reply by ghubstan** [reply]
Maybe `takeofferfees`  is not so good.  I need to think about this, but open to suggestions.

**Reply by pmartincalvo** [reply]
I dislike `Take` because you are not really taking. And I dislike `Fees` because it's not only fees, since there is also the security deposit.

I guess the common root of all things involved is the need for the user to lock funds, with some potentially coming back and some not under different scenarios (completed trade, canceled trade, trade going under arbitration with a loss of deposit for the user). Would something like `GetTakingFunds` sound accurate?

Another thing I didn't think is: how to deal with the BTC/BSQ optionality for the trading fee. I guess the easy solution is to show both?

**Reply by ghubstan** [reply]
`GetTakingFunds` seems accurate.  Something like (CLI):    `gettakingfunds --offer-id=[offer-id]`
And the output should show trading fees for paying them btc or bsq.

**Reply by pmartincalvo** [reply]
Great, glad you agree.

As well as miner fees and the security deposit amount, right?

**Reply by ghubstan** [reply]
Right.

**Reply by pmartincalvo** [reply]
@ghubstan is there a PR or Issue where we can keep a track of the status of this? Thanks.

**Reply by ghubstan** [reply]
I will create a PR for this work -- probably next week.

**Reply by ghubstan** [reply]
I will reference this discussion in it.

**Reply by ghubstan** [reply]
I still need to finish my current task before I begin work on this, but it is next on my plate.

**Reply by pmartincalvo** [reply]
Great :)

**Comment by ghubstan** [comment]
Other team members have decided to significantly reduce the frequency of Bisq v1 updates (new releases are bug-fixes only), to focus on Bisq v2.  

Work on any new Bisq v1 API methods is on indefinite hold.

**Reply by pmartincalvo** [reply]
:(
